yt_dlp/extractor/ard.py

```diff
@@ -601,3 +603,170 @@ def fetch_page(page_num):
+                'timestamp': 1735629900,
+                'series': 'Auf der Spur – Die ARD Ermittlerkrimis',
+                'channel': 'ARD',
+                'episode': 'Episode 1',
+                'episode_number': 1,
+            },
+        },
+        {
+            'url': 'https://www.ardaudiothek.de/episode/urn:ard:extra:d2fe7303d2dcbf5d/',
+            'info_dict': {
+                'id': 'urn:ard:extra:d2fe7303d2dcbf5d',
+                'ext': 'mp3',
+                'title': 'Trailer: Fanta Vier Forever, Baby!?!',
+                'description': 'md5:b64a586f2e976b8bb5ea0a79dbd8751c',
+                'channel': 'SWR',
+                'duration': 62,
+                'thumbnail': 'https://api.ardmediathek.de/image-service/images/urn:ard:image:48d3c255969be803',
+                'series': 'Fanta Vier Forever, Baby!?!',
+                'timestamp': 1732108217,
+                'upload_date': '20241120',
+            },
+        },
+    ]
+
+    _QUERY_ITEM = '''\
+    query($id: ID!) {
+        item(id: $id) {
+            audioList {
+                href
+                distributionType
+                audioBitrate
+                audioCodec
+            }
+            show {
+              title
+            }
+            image {
+              url1X1
+            }
+            programSet {
+              publicationService {
+                organizationName
+              }
+            }
+            description
+            title
+            duration
+            startDate
+            episodeNumber
+        }
+    }'''
+
+    def _real_extract(self, url):
+        urn = self._match_id(url)
+        item = self._graphql_query(urn, self._QUERY_ITEM)['item']
+        return {
+            'id': urn,
+            **traverse_obj(item, {
+                'formats': ('audioList', lambda _, v: url_or_none(v['href']), {
+                    'url': 'href',
+                    'format_id': ('distributionType', {str}),
+                    'abr': ('audioBitrate', {int_or_none}),
+                    'acodec': ('audioCodec', {str}),
+                    'vcodec': {value('none')},
+                }),
+            }),
+            **traverse_obj(item, {
+                'channel': ('programSet', 'publicationService', 'organizationName', {str}),
+                'description': ('description', {str}),
+                'duration': ('duration', {int_or_none}),
+                'series': ('show', 'title', {str}),
+                'episode_number': ('episodeNumber', {int_or_none}),
+                'thumbnail': ('image', 'url1X1', {url_or_none}, {update_url(query=None)}),
+                'timestamp': ('startDate', {parse_iso8601}),
+                'title': ('title', {str}),
+            }),
+        }
+
+
+class ARDAudiothekPlaylistIE(ARDAudiothekBaseIE):
+    _VALID_URL = r'''https:?//(?:www\.)?ardaudiothek\.de/sendung/(?P<playlist>[a-zA-Z-]+)/(?P<id>urn:ard:show:[a-f0-9]{16})'''
```

Comment:
I changed this to `\w` for a reason. Otherwise, URLs such as https://www.ardaudiothek.de/sendung/100-berlin/urn:ard:show:4d248e0806ce37bc/ would not match. Maybe add that as a `only_matching` test.
```suggestion
    _VALID_URL = r'''https:?//(?:www\.)?ardaudiothek\.de/sendung/(?P<playlist>[\w-]+)/(?P<id>urn:ard:show:[a-f0-9]{16})'''
```

Reply:
Ah good point, must have missed this in your first suggestion.